frontend/src/pages/ArticleDetails.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+import { Container, Typography, Card, CardContent, CardMedia, Box } from '@mui/material';
+
+function ArticleDetails() {
+  const location = useLocation();
+  const { article } = location.state || {}; // Get article from state
```

Comment:
`location.state` is usually used for less data.

Instead of fetching from `location.state` we should either call a separate API to fetch the article details or get the details from global state. We already know that we are adding comments now so there should be a separate API.

The article list API should only fetch the data that is displayed on the list view. Not complete information. Why return everything when a few attributes are displayed?